conans/cli/cli.py

```diff
@@ -28,6 +29,7 @@ def __init__(self, conan_api):
             type(conan_api))
         self._conan_api = conan_api
         self._out = conan_api.out
+        self._cli_out = CliOutput(sys.stdout, self._out.color)
```

Comment:
Maybe the idea would be to use the same pattern we might use for the application sys.stderr:

- Can we assume that we are always using sys.stdout and sys.stderr, and avoid passing them as arguments?
- It is possible and easy to redirect them for testing purposes.
- Maybe it is necessary to control the output and interactions with the progress bars?

Reply:
It would also be good if the names could be a bit more different for the app logs around sys.stderr and the output around sys.stdout. Is it ``CliOutput`` and ``ConanOutput``. Maybe they are too similar? How are the instances going to be called in the codebase, ``cli_out`` and ``conan_out``? Or can be the second just ``out``? I can't think of better naming right now...

Reply:
Answering here but doing changes in #7502 that is the base for this one. I'll do the changes related to `create`, `upload` and piping here.

> * Can we assume that we are always using sys.stdout and sys.stderr, and avoid passing them as arguments?
I think for CliOutput and `stderr` yes but for ConanOutput could it make sense that users of just ConanAPI want to redirect to `stdout`?
> * It is possible and easy to redirect them for testing purposes.
I have to check that.
> * Maybe it is necessary to control the output and interactions with the progress bars?
I added a handler for tqdm. Progress bars will be shown in stderr but not in the logger file por example: https://github.com/conan-io/conan/pull/7502#issuecomment-671959227

Reply:
> It would also be good if the names could be a bit more different for the app logs around sys.stderr and the output around sys.stdout. Is it `CliOutput` and `ConanOutput`. Maybe they are too similar? How are the instances going to be called in the codebase, `cli_out` and `conan_out`? Or can be the second just `out`? I can't think of better naming right now...

I think that `cli_out` and `out` could be fine. I'll change to that in #7502